### PR TITLE
Provide defaults for more presubmit config fields.

### LINKED ISSUE #6655: agent: kubernetes should be default for prow jobs?
aka, we don't have and won't have any `agent: jenkins` anymore

cc @cjwagner @BenTheElder

### PR DESCRIPTION
**Pull request by cjwagner**
Specifically,
- Default `Trigger` and `RerunCommand` to match `/test <job-name>`.
- Default `Context` to the the job's `Name`.
- Default `Agent` to `kubernetes`.

This should help us passively promote some standards and make it simpler for new Prow users to add new presubmit configurations.

I'll make a separate PR to remove these fields from our config files where the defaults are appropriate (i.e. everywhere). We should wait to merge the other PR until this PR merges and is deployed to avoid config loading failures.

fixes #6655 
/area jobs
/area prow
/cc @krzyzacy @BenTheElder @stevekuznetsov

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Do we have a script for automated migration/culling of existing fields or identifying which existing fields cannot simply be dropped since they are non-standard?

**Comment by stevekuznetsov** [comment]
@petr-muller FYI

File: prow/config/config.go [hunk 1]
```diff
@@ -515,18 +515,22 @@ func (c *Config) finalizeJobConfig() error {
 		}
 	}
 
-	// Ensure that regexes are valid.
+	// Ensure that regexes are valid and set defaults.
```
Comment (stevekuznetsov) [inline prow/config/config.go]: We need to doc this on the `prow/README.md`

**Comment by cjwagner** [comment]
> Do we have a script for automated migration/culling of existing fields or identifying which existing fields cannot simply be dropped since they are non-standard?

No, I just used `find` and `sed`, but all of our contexts already matched the job names, the trigger regexps and rerun commands all matched `/test <job-name`, and all our jobs use the `kubernetes` agent so we already used the defaults everywhere.

**Pull request merged**